// Copyright 2020 Lisandro Bravo.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

message MQTT_MESSAGE_TRANSITION{
  required string topic = 1;
  required string payload = 2;
}

message actionMap {
  required int32 start_action = 1;
  required int32 end_action = 2;
  required MQTT_MESSAGE_TRANSITION mqtt_message = 3;
}



// Options to NonMaxSuppression calculator, which performs non-maximum
// suppression on a set of detections.
message transitionDynamicGesturesCalculatorOptions {
  extend CalculatorOptions {
    optional transitionDynamicGesturesCalculatorOptions ext = 55483322;
  }
  
  
  // Maximum time allowed between startind and endind gesture .
  // in secconds
  optional double time_out_s = 1 [default = 2.500];
  
  repeated actionMap actions_map = 2;
}
